projects/cdk/directives/auto-focus/autofocus.directive.ts

```diff
@@ -84,12 +86,33 @@ export class TuiAutoFocusDirective implements AfterViewInit {
         return this.element.matches('input, textarea');
     }
 
+    private isFullScreenDialog() {
+        return (
+            this.element.closest('[data-size="page"]') != null ||
+            this.element.closest('[data-size="fullscreen"]') != null
+        );
+    }
+
     private iosWebkitAutofocus() {
+        const animationMinDelay = 100;
         const fakeInput: HTMLElement = this.renderer.createElement('input');
 
         fakeInput.style.position = 'absolute';
         fakeInput.style.opacity = '0';
-        fakeInput.style.height = '0';
+
+        /**
+         * @note: emulate original container size
+         */
+        fakeInput.style.height = px(this.element.clientHeight);
+        fakeInput.style.width = px(this.element.clientWidth);
+
+        /**
+         * @note:
+         * Move the focus point where the focus will browser navigate when the virtual keyboard opened
+         */
+        const offsetY = this.isFullScreenDialog() ? 0 : VIRTUAL_KEYBOARD_HEIGHT;
```

Comment:
Doesn't it change from type to type? Even if it doesn't, this is a hardcode for a particular version of iOS virtual keyboard which can change.

Reply:
removed it